client/components/data/store-connection/index.jsx

```diff
@@ -58,17 +58,22 @@ const StoreConnection = React.createClass( {
 	},
 
 	render() {
+		const props = {
+			...this.props,
```

Comment:
This is something that could probably go into a different PR, but I noticed some strange behaviors in the store connector and I felt that this and a couple of other changes would clean it up quite a bit.

By including the props here it means we don't have to duplicate all of the props in `getPropsFromState` or whatever it is that this calls to generate props. 

After some investigation I realized that about half of this file could be zapped anyway.